Data_Structures_Answers.md

```diff
@@ -1,18 +1,30 @@
 Add your answers to the questions below.
 
-1. What is the runtime complexity of your ring buffer's `append` method?
+1. What is the runtime complexity of your ring buffer's `append` method? 
+
+    1a. O(n)
 
 2. What is the space complexity of your ring buffer's `append` function?
 
+    2a. O(1)
+
 3. What is the runtime complexity of your ring buffer's `get` method?
 
+    3a. O(n)
+
 4. What is the space complexity of your ring buffer's `get` method?
 
+    4a. O(1)
 
 5. What is the runtime complexity of the provided code in `names.py`?
 
+    5a. O(n^2)
+
 6. What is the space complexity of the provided code in `names.py`?
+    6a. O(n)
 
 7. What is the runtime complexity of your optimized code in `names.py`?
+    7a. O(log n)
 
 8. What is the space complexity of your optimized code in `names.py`?
+    8a. O(n)
```

Comment:
Overall, good answers. However, without the explantations, I have little reference of knowing how accurate your analysis of your own answers are.